src/prefect/utilities/templating.py

```diff
@@ -195,13 +196,20 @@ def apply_values(
         raise ValueError(f"Unexpected template type {type(template).__name__!r}")
 
 
+def replace_with_value(placeholder: str, value: Any) -> Any:
+    """A block reference replacement function that returns the value unchanged."""
+    return value
+
+
 @inject_client
 async def resolve_block_document_references(
-    template: T, client: Optional["PrefectClient"] = None
+    template: T,
+    replacement_function: Callable[[str, Any], Any] = replace_with_value,
```

Comment:
Slight preference for `value_transformer` as the name for this parameter. Also, it should probably default to `None` since functions are technically mutable, and we might want to put this after `client` to not disturb the current parameter order.
```suggestion
    value_transformer: Optional[Callable[[str, Any], Any]] = None,
```

Reply:
Good thoughts! Done!